react/features/mobile/callkit/middleware.js

```diff
@@ -286,23 +286,14 @@ function _onPerformEndCallAction({ callUUID }) {
  * {@code performSetMutedCallAction}.
  * @returns {void}
  */
-function _onPerformSetMutedCallAction({ callUUID, muted: newValue }) {
+function _onPerformSetMutedCallAction({ callUUID, muted }) {
     const { dispatch, getState } = this; // eslint-disable-line no-invalid-this
     const conference = getCurrentConference(getState);
 
     if (conference && conference.callUUID === callUUID) {
-        // Break the loop. Audio can be muted from both CallKit and Jitsi Meet.
-        // We must keep them in sync, but at some point the loop needs to be
-        // broken. We are doing it here, on the CallKit handler.
-        const tracks = getState()['features/base/tracks'];
-        const oldValue = isLocalTrackMuted(tracks, MEDIA_TYPE.AUDIO);
-
-        newValue = Boolean(newValue); // eslint-disable-line no-param-reassign
-
-        if (oldValue !== newValue) {
-            sendAnalytics(createTrackMutedEvent('audio', 'callkit', newValue));
-            dispatch(setAudioMuted(newValue, /* ensureTrack */ true));
-        }
+        muted = Boolean(muted); // eslint-disable-line no-param-reassign
+        sendAnalytics(createTrackMutedEvent('audio', 'callkit', muted));
+        dispatch(setAudioMuted(muted, /* ensureTrack */ true));
```

Comment:
I'm not sure if it's good idea to not do any check here and just dispatch setAudioMuted actions even if we could be matching the state already

Reply:
I thought the same, but then I asked myself: "how and when can they go out of sync?" and I couldn't find a way, so I thought that being too defensive would make the code harder to read because the reader would ask herself the same question.

Reply:
We've discussed it offline and because it will only modify the base/media which is the intention and in fact will work very similar to the current mute button on the toolbar. It's fine to not do this check here given that we won't be getting this callback for our own action (in contrary to what was happening before this PR).